lib/cog.ex

```diff
@@ -18,13 +18,18 @@ defmodule Cog do
     case Supervisor.start_link(children, opts) do
       {:ok, top_sup} ->
         # Verify the latest schema migration after starting the database worker
-        {sm_status, sm_message} = verify_schema_migration()
-        log_message(sm_status, sm_message)
-        if sm_status == :error do
-          abort_cog()
-        else
-          {:ok, top_sup}
-        end
+        #
+        # NOTE: This _may_ cause issues in the future, since we
+        # currently announce and install the embedded bundle during
+        # startup (i.e., before doing the schema migration check). If
+        # at some point we were to automatically upgrade the embedded
+        # bundle in such a way that required a database migration
+        # beforehand, the resulting error might be confusing.
+        #
+        # Consider doing this in a process that runs after the Repo
+        # comes up, but before anything else is done
+        :ok = verify_schema_migration!
```

Comment:
If we raise here due to the migrations not being up to date, does that still stop the supervisor from starting up? Also, how does that change the error message printed?